packages/contracts/contracts/Unirep.sol

```diff
@@ -392,13 +391,11 @@ contract Unirep is IUnirep, VerifySignature {
     function verifyEpochKeyProof(
         uint256[] memory publicSignals,
         uint256[8] memory proof
-    ) public returns (bool) {
+    ) public view returns (bool) {
```

Comment:
Turn this to a `view` function because I think it might duplicated if user verify proof and submit attestations
And a `view` function might be more useful if the backend or frontend wants to verify proofs and they don't have to send another transactions

Reply:
Okay I understand
so maybe in this case it doesn't have to return a `true` or `false`, it can revert the transaction?
I don't know if there is any case that a `false` proof can execute a valid transaction.
and it will be better to be tested 
(since `ethers` might not get the return value of a transaction?https://ethereum.stackexchange.com/questions/88119/i-see-no-way-to-obtain-the-return-value-of-a-non-view-function-ethers-js)

> verify the roots using the offchain data

yeah I agree with this

Reply:
>  I don't know if there is any case that a false proof can execute a valid transaction.
Yea I tend to agree. If an application wants to take this as a boolean they can use a `try`/`catch` statement.